### PR TITLE
Backport #119 to shardtree v0.3.x

### PR DESCRIPTION
**Pull request by nuttycom**
Backport of #119

### CODE REVIEW

**Review by daira — APPROVED** [review]
utACK. (I think @str4d might have said he had another change to make.)

File: shardtree/src/lib.rs [hunk 1]
```diff
@@ -1103,7 +1109,8 @@ impl<
             cur_addr = cur_addr.parent();
         }
 
-        Ok(MerklePath::from_parts(witness, position).unwrap())
+        Ok(MerklePath::from_parts(witness, position)
+            .expect("witness has length DEPTH because we extended it to the root"))
```
Comment (daira) [inline shardtree/src/lib.rs:1113]: That depends on collision resistance, but I'm fine with using `.expect` in that case.

**Review by daira — APPROVED** [review]
utACK 392e395fdedebcb6cbfbae33d445f1aa2b36f482

**Review by str4d — APPROVED** [review]
utACK 77fa60088fd7bd65e9c4af73864dc67d6e52ae9b. `unwrap`s now all document why they are there. Several of them can be reachable if the caller makes a programming error; this can only be fixed with a subsequent breaking API change (as in my other PRs).

**Pull request merged**